virtcontainers/kata_agent.go

```diff
@@ -673,7 +673,10 @@ func (k *kataAgent) listInterfaces() ([]*vcTypes.Interface, error) {
 	}
 	resultInterfaces, ok := resultingInterfaces.(*grpc.Interfaces)
 	if ok {
-		return k.convertToInterfaces(resultInterfaces.Interfaces), err
+		ifaces, err := k.convertToInterfaces(resultInterfaces.Interfaces)
+		if err == nil {
+			return ifaces, nil
+		}
 	}
 	return nil, err
```

Comment:
@dgibson Didnt get a chance to get to this PR earlier.
If the intention was to capture the error from `convertToInterfaces`, I am afraid you are not quite doing that here, since the err on line 677  has scope only within the if block and not outside it. So line 681 will always return nil for err.
You could just change the previous return statement to this `return k.convertToInterfaces(resultInterfaces.Interfaces)` for things to work correctly.

Reply:
Drat, good point.  The interaction of `:=` with variable shadowing is a real Go gotcha :/.

Reply:
Filed #3032 and #3033 to address this.